### PR TITLE
Bug 1929277: Adjust priority class of core platform prometheus

### PR DESCRIPTION
**Pull request by lilic**
* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: test/e2e/priority_class_test.go [hunk 1]
```diff
@@ -45,4 +50,10 @@ func TestToEnsureUserPriorityClassIsPresentAndLower(t *testing.T) {
 	if userPriorityClass.Value >= systemClusterPriorityClass.Value || userPriorityClass.Value >= systemNodePriorityClass.Value {
 		t.Fatalf("openshift-user-critical was higher priority than existing classes: %d", userPriorityClass.Value)
 	}
+	// Ensure openshift-system-critical is a lower value than the system class' ones, but higher than openshift-user-critical.
+	if openshiftSystemPriorityClass.Value < systemClusterPriorityClass.Value ||
+		openshiftSystemPriorityClass.Value < systemNodePriorityClass.Value ||
+		userPriorityClass.Value < openshiftSystemPriorityClass.Value {
+		t.Fatalf("openshift-system-critical was wrong priority class: %d", openshiftSystemPriorityClass.Value)
+	}
```
Comment (s-urbaniak) [inline test/e2e/priority_class_test.go:58]: :tada:

**Comment by lilic** [comment]
Aimed towards master, but I guess 4.7 is better as its urgent?

**Comment by s-urbaniak** [comment]
let's give @smarterclayton and @rphillips the final call (to unhold)

**Comment by lilic** [comment]
thanks, that is even better Serg! 🤞

**Comment by rphillips** [comment]
BZ for this PR https://bugzilla.redhat.com/show_bug.cgi?id=1913532

**Comment by lilic** [comment]
/retest

from what I can tell its non monitoring issue.

**Comment by lilic** [comment]
/retest

> Cluster operator authentication Degraded is True with APIServerDeployment_PreconditionNotFulfilled::IngressStateEndpoints_MissingSubsets::O

**Comment by bparees** [comment]
/hold cancel

not seeing anything in the slack discussions that makes me think @rphillips or @smarterclayton disagree w/ this.

File: manifests/0000_50_cluster-monitoring-operator_00_0openshift-system-priority-class.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: openshift-system-critical
+  annotations:
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+value: 1000001000
```
Comment (rphillips) [inline manifests/0000_50_cluster-monitoring-operator_00_0openshift-system-priority-class.yaml:8]:
```
The PriorityClass "openshift-system-critical" is invalid: value: Forbidden: maximum allowed value of a user defined priority is 1000000000
```

**Comment by bparees** [comment]
/hold

this needs to be redone based on https://github.com/openshift/cluster-monitoring-operator/pull/1060

**Comment by simonpasquier** [comment]
superseded by https://github.com/openshift/cluster-monitoring-operator/pull/1067

**Pull request closed without merging**